row_cache.hh

```diff
@@ -362,15 +362,29 @@ public:
     // User needs to ensure that the row_cache object stays alive
     // as long as the reader is used.
     // The range must not wrap around.
+
+    // this method's overloading used in tests only as far as they don't use tombstone_gc_state
+    flat_mutation_reader_v2 make_reader(schema_ptr s,
+                                     reader_permit permit,
+                                     const dht::partition_range& range,
+                                     const query::partition_slice& slice,
+                                     const io_priority_class& pc = default_priority_class(),
+                                     tracing::trace_state_ptr trace_state = nullptr,
+                                     streamed_mutation::forwarding fwd = streamed_mutation::forwarding::no,
+                                     mutation_reader::forwarding fwd_mr = mutation_reader::forwarding::no) {
+        return make_reader(s, permit, range, slice, nullptr, pc, trace_state, fwd, fwd_mr);
+    }
+
     flat_mutation_reader_v2 make_reader(schema_ptr s,
                                      reader_permit permit,
                                      const dht::partition_range& range,
                                      const query::partition_slice& slice,
+                                     const tombstone_gc_state* gc_state,
```

Comment:
You could add this as a last parameter and default to nullptr. This way you don't need to add an overload.

Reply:
It was the first thing I did but it doesn't help unfortunately: some tests pass own values and don't use default ones so either we make changes in many tests or overload the method. I chose the simplest way )